torchao/prototype/float8nocompile/float8nocompile_scaling_utils.py

```diff
@@ -59,3 +61,60 @@ def hp_tensor_to_float8nocompile_dynamic(
         gemm_input_role,
         None,
     )
+
+
+class Float8NoCompileConversionFunc(torch.autograd.Function):
+    """
+    A differentiable conversion to fp8.
+    * forward: convert from high precision to float8
+    * backward: pass the gradient without changes
+    """
+
+    @staticmethod
+    def forward(
+        ctx,
+        tensor: torch.Tensor,
+        float8_dtype: torch.dtype,
+        linear_mm_config: LinearMMConfig,
+        gemm_input_role: GemmInputRole,
+    ):
+        return triton_hp_tensor_to_float8_dynamic(
+            tensor,
+            float8_dtype,
+            linear_mm_config,
+            gemm_input_role,
+        )
+
+    @staticmethod
+    def backward(ctx, g):
+        return g, None, None, None, None, None
+
+
+class NoopFwToFloat8NoCompileBwDynamic(torch.autograd.Function):
+    """
+    A differentiable conversion to fp8.
+    * forward: no-op
+    * backward: convert to fp8_e5m2 with tensor-wise dynamic scaling
```

Comment:
nit: remove e5m2 from docstring, since the target dtype is passed as an argument